app/models/assignment_repo.rb

```diff
@@ -6,14 +6,36 @@ class AssignmentRepo < ActiveRecord::Base
 
   belongs_to :assignment
   belongs_to :repo_access
+  belongs_to :user
 
   validates :assignment, presence: true
 
   validates :github_repo_id, presence:   true
   validates :github_repo_id, uniqueness: true
 
-  validates :repo_access, presence:   true
-  validates :repo_access, uniqueness: { scope: :assignment }
+  before_validation(on: :create) do
+    if organization
+      create_github_repository
+      push_starter_code
+      add_user_as_collaborator
+    end
+  end
+
+  before_destroy :destroy_github_repository
+
+  # Public
+  #
+  def add_user_as_collaborator
+    github_user = GitHubUser.new(user.github_client, user.uid)
+    repository  = GitHubRepository.new(organization.github_client, github_repo_id)
+
+    begin
```

Comment:
This is the same pattern we used with pushing starter code. Is there a way to de-duplicate and share the behavior?